adal/authentication_context.py

```diff
@@ -67,8 +70,20 @@ def __init__(self, authority, validate_authority=None, cache=None):
         self._oauth2client = None
         self.correlation_id = None
         env_value = os.environ.get('ADAL_PYTHON_SSL_NO_VERIFY')
+        if api_version is not None:
+            warnings.warn(
+                """The default behavior of including api-version=1.0 on the wire
+                is now deprecated.
+                Future version of ADAL will change the default value to None.

```

Comment:
That makes sense. And it seems an easy fix at this point would be changing the warning type to `DeprecationWarning`, which will then be ignored by default, I've tested it by running `python -c "import adal; adal.AuthenticationContext('https://login.windows.net/tenant')"` on both Python 2 and Python 3.

I've also add docstring for the new parameter. Please review [the latest commit](https://github.com/AzureAD/azure-activedirectory-library-for-python/pull/57/commits/791c7878a85d41e641310037992def12171d1bd6)